docs/src/userguide/extension_types.rst

```diff
@@ -1001,6 +1001,57 @@ disable this auto-generation and can be used to support pickling of more
 complicated types.
 
 
+Pattern matching and extension types (Python 3.10+)
+===================================================
+
+Python 3.10 introduced the structural pattern matching feature
+(documented in `the Python documentation
+<https://docs.python.org/3/reference/compound_stmts.html#the-match-statement>`_,
+and `PEP 634 <https://peps.python.org/pep-0634>`_).
+where Python objects can be decomposed if they meet a particular 
+set of rules. For a type to be treated as a sequence in pattern 
+matching it must have the flag ``Py_TPFLAGS_SEQUENCE`` in its 
+internal "type flags", and similarly for a type to be treated 
+as a mapping it must have ``Py_TPFLAGS_MAPPING`` set.
+
+There's a number of ways for these flags to be set: they're 
+set for a number of builtin types.  They will be set on a
+type where one of the base classes has the flag (including
+the builtin types, and also the relevant abstract base
+class in ``abc.collections``).
```

Comment:
Minor nit: it's `collections.abc`, not the other way around.

Reply:
Thanks @TeamSpen210